File: src/main/java/no/nav/fo/veilarbregistrering/enhet/EnhetGateway.java
package no.nav.fo.veilarbregistrering.enhet;

import no.nav.fo.veilarbregistrering.arbeidsforhold.Organisasjonsnummer;

import java.util.Optional;

public interface EnhetGateway {

    Optional<Organisasjonsdetaljer> hentOrganisasjonsdetaljer(Organisasjonsnummer organisasjonsnummer);
}

File: src/main/kotlin/no/nav/fo/veilarbregistrering/enhet/EnhetGateway.kt
package no.nav.fo.veilarbregistrering.enhet

import no.nav.fo.veilarbregistrering.arbeidsforhold.Organisasjonsnummer
import java.util.*

fun interface EnhetGateway {
    fun hentOrganisasjonsdetaljer(organisasjonsnummer: Organisasjonsnummer): Organisasjonsdetaljer?
}